### PR TITLE
Bump WEEK_12 compat to v1.3.0

### PR DESCRIPTION
**Pull request by apivovarov**
Bump 12W compat to v1.3.0 released on July 16, 2024

Tags: https://github.com/openxla/stablehlo/tags

12W is used in `xla::GetDefaultStablehloVersion()` which is used by `PjRtCApiCompiler::Compile()` xla/pjrt/pjrt_c_api_client.cc:2317

### CODE REVIEW

File: stablehlo/dialect/Version.cpp [hunk 1]
```diff
@@ -85,7 +85,7 @@ Version Version::fromCompatibilityRequirement(
     case CompatibilityRequirement::WEEK_4:
       return Version(1, 7, 0);  // v1.7.0 - Sept 05, 2024
     case CompatibilityRequirement::WEEK_12:
-      return Version(1, 1, 0);  // v1.1.0 - May 30, 2024
+      return Version(1, 3, 0);  // v1.3.0 - Jul 10, 2024
```
Comment (GleasonK) [inline stablehlo/dialect/Version.cpp:88]: Per the comment above, we use the date a release was tagged, which was July 16th:

image

Still 12w ago (actually 12w perfectly 🎉 ) but could you update the comment to use the tag date?

**Review by GleasonK — APPROVED** [review]

File: stablehlo/dialect/Version.cpp (hunk 1 above)
Comment (apivovarov) [reply stablehlo/dialect/Version.cpp]: fixed

**Comment by apivovarov** [comment]
Seems that v1.4.0 was tagged on the same day as v1.3.0 (on July 16, 2024).
https://github.com/openxla/stablehlo/tags?after=v1.5.0

Should we return 1.4.0 for WEEK_12 req?

**Comment by apivovarov** [comment]
For convenience we can use the following script to get WEEK_4 and WEEK_12 tag / tag date / days ago

```bash
python3 get_compatibility_versions.py 

WEEK_4: (tag: v1.7.1) 2024-09-06T15:27:44-07:00 34 days ago
WEEK_12: (tag: v1.4.0) 2024-07-15T14:21:55-07:00 87 days ago
```

```py
import subprocess
import time

utc = int(time.time())

command = [
  "git",
  "log",
  "--tags",
  "--simplify-by-decoration",
  "--pretty=format:%ct,%d,%cI",
]
process = subprocess.Popen(
  command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True
)

WEEK_4_tag = ""
WEEK_12_tag = ""

for line in process.stdout:
  line = line.strip().split(",")
  tag_sec = int(line[0])
  diff_days = int((utc - tag_sec) / (60 * 60 * 24))
  if WEEK_4_tag == "" and diff_days >= 7 * 4:
    WEEK_4_tag = line[1].strip()
    print("WEEK_4:", WEEK_4_tag, line[2], diff_days, "days ago")
  if WEEK_12_tag == "" and diff_days >= 7 * 12:
    WEEK_12_tag = line[1].strip()
    print("WEEK_12:", WEEK_12_tag, line[2], diff_days, "days ago")
    break
```

Maybe add it to one of the documents?

**Review by GleasonK — APPROVED** [review]

**Pull request merged**